apps/prairielearn/src/models/course-instance-usages.sql

```diff
@@ -74,6 +74,11 @@ FROM
   LEFT JOIN course_instances AS ci ON (ci.course_id = a.course_instance_id)
 WHERE
   gj.id = $grading_job_id
+  -- Avoid inserting anything if we'd compute a NULL duration.
+  AND gj.grading_received_at IS NOT NULL
+  AND gj.grading_finished_at IS NOT NULL
+  -- Avoid inserting negative durations.
+  AND gj.grading_finished_at > gj.grading_received_at
```

Comment:
For completeness, we'll also reject negative durations. As noted on https://github.com/PrairieLearn/PrairieLearn/pull/11448#pullrequestreview-2651218816, there are actually instances of this in practice.